libs/wingc/src/diagnostic.rs

```diff
@@ -6,7 +7,7 @@ use crate::debug;
 pub type FileId = String;
 pub type CharacterLocation = Point;
 pub type ByteIndex = usize;
-pub type Diagnostics = Vec<Diagnostic>;
+pub type Diagnostics = IndexSet<Diagnostic>;
```

Comment:
What lead to me adding this is that I now call `resolve_reference` in some cases more than once on the same reference in the type-checker. Specifically when type checking a call expression I first type check the expression that evaluates to the function type (usually a reference to a function/method name) and then later I call `resolve_reference` again just to check if that method/function is static or not so I know if I should expect a `this` first arg to it or not.
The double call to `resolve_reference` can lead to duplicate "unknown symbol" errors.

But then I noticed that some other duplicates where removed from our invalid snapshot tests, so I guess there were other scenarios that caused duplicates. It makes sense because there's no way for me to know if some function in our compiler produces an error and I don't want to limited to not calling functions more than once on the same inputs.